webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/UploadService.scala

```diff
@@ -138,29 +154,43 @@ class UploadService @Inject()(dataSourceRepository: DataSourceRepository,
             tempFile.write(bytes)
             tempFile.close()
           }
+          Fox.successful(())
         } catch {
           case e: Exception =>
             runningUploadMetadataStore.removeFromSet(redisKeyForFileChunkSet(uploadId, filePath),
                                                      String.valueOf(currentChunkNumber))
-            val errorMsg = s"Error receiving chunk $currentChunkNumber for upload ${datasourceId.name}: ${e.getMessage}"
+            val errorMsg = s"Error receiving chunk $currentChunkNumber for upload ${dataSourceId.name}: ${e.getMessage}"
             logger.warn(errorMsg)
-            return Fox.failure(errorMsg)
+            Fox.failure(errorMsg)
         }
-      case false => ()
-    }
+      } else Fox.successful(())
+  }
+
+  def cancelUpload(cancelUploadInformation: CancelUploadInformation): Fox[Unit] = {
+    val uploadId = cancelUploadInformation.uploadId
+    for {
+      dataSourceId <- getDataSourceIdByUploadId(uploadId)
+      knownUpload <- isKnownUpload(uploadId)
+    } yield
+      if (knownUpload) {
+        logger.info(f"Canceling dataset upload of ${dataSourceId.team}/${dataSourceId.name} with id ${uploadId}...")
+        PathUtils.deleteDirectoryRecursively(uploadDirectory(dataSourceId.team, uploadId))
+        removeFromRedis(uploadId)
```

Comment:
please make sure that removeFromRedis happens first, so that a new incoming chunk does not try to write to a deleted file